framework/archiver/utils.py

```diff
@@ -0,0 +1,79 @@
+from framework.auth import Auth
+
+from framework.archiver import (
+    StatResult, AggregateStatResult,
+    ARCHIVER_FAILURE,
+)
+from framework.archiver.settings import (
+    ARCHIVE_PROVIDER,
+)
+
+def archive_provider_for(node, user):
+    return node.get_addon(ARCHIVE_PROVIDER)
+
+def has_archive_provider(node, user):
+    return node.has_addon(ARCHIVE_PROVIDER)
+
+def link_archive_provider(node, user):
+    addon = node.get_or_add_addon(ARCHIVE_PROVIDER, auth=Auth(user))
+    addon.on_add()
+    node.save()
+
+def catch_archive_addon_error(node, addon_short_name, errors=[]):
+    node.archived_providers[addon_short_name].update({
+        'status': ARCHIVER_FAILURE,
+        'errors': errors,
+    })
+    node.save()
+
+def delete_registration_tree(node):
+    node.is_deleted = True
+    node.save()
+    [delete_registration_tree(child) for child in node.nodes]
+
+def archive_folder_name(addon):
+    """Create a name for an archived addon's folder
+
+    :param addon: AddonNodeSettings instance of addon in question
+    """
+    name = "Archive of {addon}".format(addon=addon.config.full_name)
+    folder_name = (getattr(addon, 'folder_name') or '').lstrip('/').strip()
+    if folder_name:
+        name = name + ": {folder}".format(folder=folder_name)
```

Comment:
Addons should have a name field or something that is used to create their top level folders.
```
s3.name  # S3: {bucketname}
db.name  # DropBox: {folder}
...
```